scaladoc/test/dotty/tools/scaladoc/signatures/SignatureTest.scala

```diff
@@ -84,10 +84,10 @@ abstract class SignatureTest(
 
   private def findName(signature: String, kinds: Seq[String]): Option[String] =
     for
```

Comment:
I meant something like:

```scala
for
  kindMatch <- kinds.flatMap(k => s"\\b$k\\b".r.findFirstMatchIn(signature)).headOption
  kind <- Option(kindMatch.group(0)) // to filter out nulls
  afterKind <- Option(kindMatch.after(0)) // to filter out nulls
  name <- if kind.contains("extension") then Some(signature)
                else identifierRegex.findFirstMatchIn(afterKind).map(_.group(1))
yield name
```

I think that the reason the tests pass now is that for `extension` signatures, `identifierRegex.findFirstMatchIn(afterKind)` might give `None` and so those cases aren't tested.

Reply:
It is corrected, thank you very much for your help.